### PR TITLE
Fix block proposals in the REST validator client

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Bug fix

**What does this PR do? Why is it needed?**

When the REST validator client requests a block from the beacon node, it always tries to unmarshal it into a full block. The correct thing to do is to request the block from the `/v3/validator/blocks` endpoint and read the `ExecutionPayloadBlinded` field in the response to determine whether the requested block is blinded or not. This allows us to propose blinded blocks. If a node does not support the `v3` endpoint, we do the following:
- execute `v1/validator/blinded_blocks` and, if the request is successful, marshal the result into a blinded block
- if the previous request is unsuccessful, execute `v2/validator/blocks` and marshal the result into a full block

**Other notes for review**

- I added a new e2e test which runs both the builder and the REST VC. Without the fix this test fails.
- I modified `getGraffiti` to always return 32-byte values. The `v3` endpoint checks if graffiti has the correct length, and our current e2e setup sends a shorter graffiti.
- I updated the validator client to use structs defined next to HTTP handlers rather than the ones from API Middleware.

### CODE REVIEW

**Comment by rkapka** [comment]
Converting this to draft after speaking with an SSV dev. It might be too early to only support the `v3` endpoint at this point. I will update the PR with a fallback to `v1/validator/blinded_blocks` and `v2/blocks`.

File: validator/client/beacon-api/get_beacon_block.go [hunk 1]
```diff
@@ -5,112 +5,160 @@ import (
...
 		}
-	case "deneb":
-		jsonDenebBlockContents := shared.BeaconBlockContentsDeneb{}
-		if err := decoder.Decode(&jsonDenebBlockContents); err != nil {
-			return nil, errors.Wrap(err, "failed to decode deneb block response json")
+	case version.String(version.Capella):
+		if blinded {
+			jsonCapellaBlock := shared.BlindedBeaconBlockCapella{}
+			if err := decoder.Decode(&jsonCapellaBlock); err != nil {
+				return nil, errors.Wrap(err, "failed to decode capella block response json")
```
Comment (james-prysm) [inline validator/client/beacon-api/get_beacon_block.go:120]: can mention blinded for these

File: validator/client/beacon-api/json_rest_handler.go [hunk 2]
```diff
@@ -89,9 +89,12 @@ func decodeJsonResp(resp *http.Response, responseJson interface{}) (*apimiddlewa
 	if resp.StatusCode != http.StatusOK {
 		errorJson := &apimiddleware.DefaultErrorJson{}
 		if err := decoder.Decode(errorJson); err != nil {
-			return nil, errors.Wrapf(err, "failed to decode error json for %s", resp.Request.URL)
+			if resp.StatusCode == http.StatusNotFound {
+				errorJson = &apimiddleware.DefaultErrorJson{Code: http.StatusNotFound, Message: "Resource not found"}
+			} else {
+				return nil, errors.Wrapf(err, "failed to decode error json for %s", resp.Request.URL)
```
Comment (james-prysm) [inline validator/client/beacon-api/json_rest_handler.go:95]: can we do a dump here what was trying to decode?

Comment (rkapka) [reply validator/client/beacon-api/json_rest_handler.go:95]: I am not 100% sure about this, but I think that after `decoder.Decode` reads the input, it's not longer available for reading.

**Review by james-prysm — DISMISSED** [review]
LGTM hope we can do more testing around this.

**Review by james-prysm — APPROVED** [review]

**Pull request merged**